utils/logalloc.cc

```diff
@@ -1930,14 +1939,8 @@ static void reclaim_from_evictable(region::impl& r, size_t target_mem_in_use, is
                 llogger.debug("Unable to evict more, evicted {} bytes", used - r.occupancy().used_space());
                 return;
             }
-            if (shard_segment_pool.total_memory_in_use() <= target_mem_in_use) {
-                llogger.debug("Target met after evicting {} bytes", used - r.occupancy().used_space());
-                return;
-            }
-            if (r.empty()) {
-                return;
-            }
             if (preempt && need_preempt()) {
+                llogger.debug("reclaim_from_evictable preempted");
                 return;
```

Comment:
Why didn't this check trigger? Ah, the `break` after is_compactible caused it to be skipped.

Reply:
This check was never reached, because when the region was compactible, the eviction would be skipped (except for the first time, when we optimistically evict in advance), we would never enter the inner loop, and we would go straight to the compaction after it. Skipping the eviction was intended (that's what this patch is about), but skipping the checks was not.